packages/components/src/__future__/Tag/Tag/_docs/MigrationGuide.mdx

```diff
@@ -0,0 +1,168 @@
+
+#### Before
+
+<Canvas of={LegacyTagStories.Sentiments} />
+
+#### After
+
+<Canvas of={TagMigrationStories.SentimentsMigration} sourceState="shown"/>
+
+If required, you could use `classNameOverride` to implement the same UI. It is important that any override of the default styles still meets minimum contrast ratios to ensure the component is accessible.
+
+<Canvas of={TagMigrationStories.SentimentNone} />
+
+
+### Dismissable
+
+The `dismissable` prop is no longer available in the base `Tag` component. We have split the dismissable functionality into the [RemovableTag](/docs/components-tag-future-removabletag--docs).
+
+This increases the hit box for the dismiss button and makes the `ariaLabel` in the `removeButtonProps` a required prop. This offers a more accessible experience for user and address one of the standing [accessibility issues](https://github.com/cultureamp/kaizen-discourse/issues/101) with the legacy `Tag`.
+
+#### Before
+
+<Canvas of={LegacyTagStories.Dismissable} />
+
+#### After
+
+<Canvas of={TagMigrationStories.DismissableMigration} sourceState="shown"/>
+
+Remeber that when handling dismiss (or any destructive action), a users focus should not be lost on the page.
+
+
+### Avatars
+
+Avatars are no longer provided directly to the `Tag` as a prop. Instead they will now me passed in as children.
+
+#### Before
+
+<Canvas of={LegacyTagStories.ProfileWithAvatarProps} />
+
+#### After
+
+<Canvas of={TagMigrationStories.AvatarMigration} sourceState="shown"/>
+
+
+Note: The use of `ps-4` (or `padding-inline-start`) in the `classNameOverride` is to override the default spacing at start of the `Tag`. This will ensure thethe Avatar can contour the border radius of the `Tag`.
+
+#### Dismissable Avatars
+
+Following a similar pattern to the above, you can use the `RemovableTag` with an `Avatar` and plain text as its children to achieve a dismissable version.
+
+<Canvas of={TagMigrationStories.AvatarRemovableMigration} />
+
+### Inline
+
+Orginally the `Tag` was built with the assumption that it would always required spacing at the end. This necessitated the need for the `inline` prop to remove the margin from end when it was not required.
+
+In favour of simplifying the component, `inline` and the default `margin-inline-end` has been removed. It is recommended now to control spacing between tags to use a `gap` of 12 pixels on the container of the tags, ie: `gap-12`. In instances where more specific margins are require, opt for `margin-inline-end` and `classNameOverride` on the tag to ensure RTL support is baked in.
+
+#### Before
+
+<Canvas of={LegacyTagStories.Inline} />
+
+#### After
+
+<Canvas of={TagMigrationStories.InlineMigration} sourceState="shown"/>
+
+
+### Truncation
+
+This was original controlled by the component and set the `max-width` of a component via inline styles. This can now be easily controlled with Tailwind and was not needed in the new API.
```

Comment:
I can probably elaborate on some guidance here - ie: make sure it is screen readable and that you are not hiding important information. I don't know how realistic it is to say out and out don't do it, but could definitely elaborate on this

Reply:
Would be better to push guidance to a central doc on ellipsis and a11y guidance if we know of one. The reason I didn't go hard on this is because it wasn't raised as an a11y issue in the intopia report

Reply:
I agree. Truncation isn't an accessible pattern, so it shouldn't be recommended to be done.

Reply:
I think there room for pause on saying truncation is not accessible - there is an accessible way to do it and probably valid reasons, i.e: if someone has a long name and that's in a tag. It does need to be handled properly to pass WCAG criteria (reveal on focus or reveal on activation like click) but its not a flat no from my understanding of the standard.

Reply:
I suppose the thing I'm really latching onto is this line ([in this doc](https://www.w3.org/WAI/WCAG21/Understanding/resize-text)):

> truncation is acceptable if the component's full content is available on focus or after user activation and an indication that this information can be accessed

Granted this is above resizing containers with zoom, but I wonder if that still stand 🤔 @dougmacknz may know more / have some handy dandy docs I can refer to, but will pester him next week to confirm :)

Reply:
I think if you're going to give an example migration path, it might as well be something actually usable that meets a11y standards. Just the closest solution you can think of as a suggestion.

Otherwise I'd be looking at the docs thinking you've told me the old pattern is bad, but that the new one is just as bad.

Reply:
Yeah, the Tailwind implementation of a potentially problematic truncation is still an issue, just wasn't sure if I should be trying to solve that here or not, or just provide an example of where to start before they make it properly accessible - Maybe having the examples is a bad idea and should just leave it as why its not in the new API?

Reply:
I'm going to axe the "after"  Story and just provide some guidance on why it shouldn't be done 👍